src/ert/_c_wrappers/enkf/ecl_config.py

```diff
@@ -1,207 +1,96 @@
-
-from .config_keys import ConfigKeys
-
-
-class EclConfig(BaseCClass):
-    TYPE_NAME = "ecl_config"
-
-    _alloc = ResPrototype("void* ecl_config_alloc(config_content)", bind=False)
-    _alloc_full = ResPrototype(
-        "void* ecl_config_alloc_full(  bool, \
-                                                                            char*, \
-                                                                            ecl_grid, \
-                                                                            char*, \
-                                                                            char*)",  # noqa
-        bind=False,
-    )
-    _free = ResPrototype("void  ecl_config_free( ecl_config )")
-    _get_data_file = ResPrototype("char* ecl_config_get_data_file(ecl_config)")
-    _set_data_file = ResPrototype("void  ecl_config_set_data_file(ecl_config , char*)")
-    _validate_data_file = ResPrototype(
-        "ui_return_obj ecl_config_validate_data_file(ecl_config , char*)"
-    )
-    _get_gridfile = ResPrototype("char* ecl_config_get_gridfile(ecl_config)")
-    _set_gridfile = ResPrototype("void  ecl_config_set_grid(ecl_config, char*)")
-    _validate_gridfile = ResPrototype(
-        "ui_return_obj ecl_config_validate_grid(ecl_config, char*)"
-    )
-    _get_grid = ResPrototype("ecl_grid_ref ecl_config_get_grid(ecl_config)")
-    _get_refcase_name = ResPrototype("char* ecl_config_get_refcase_name(ecl_config)")
-    _get_refcase = ResPrototype("ecl_sum_ref ecl_config_get_refcase(ecl_config)")
-    _load_refcase = ResPrototype("void  ecl_config_load_refcase(ecl_config, char*)")
-    _validate_refcase = ResPrototype(
-        "ui_return_obj ecl_config_validate_refcase(ecl_config, char*)"
-    )
-    _has_refcase = ResPrototype("bool  ecl_config_has_refcase(ecl_config)")
-    _get_depth_unit = ResPrototype("char* ecl_config_get_depth_unit(ecl_config)")
-    _get_pressure_unit = ResPrototype("char* ecl_config_get_pressure_unit(ecl_config)")
-    _active = ResPrototype("bool ecl_config_active(ecl_config)")
-    _get_last_history_restart = ResPrototype(
-        "int ecl_config_get_last_history_restart(ecl_config)"
-    )
-    _get_num_cpu = ResPrototype("int ecl_config_get_num_cpu(ecl_config)")
-
-    def __init__(self, config_content=None, config_dict=None):
-
-        if config_content is not None and config_dict is not None:
-            raise ValueError(
-                "Error: EclConfig can not be instantiated with multiple config objects"
+from ert._c_wrappers.enkf.config_keys import ConfigKeys
+
+
+class EclConfig:
+    def __init__(
+        self,
+        data_file: Optional[str] = None,
+        grid_file: Optional[str] = None,
+        refcase_file: Optional[str] = None,
+    ):
+        self._data_file = data_file
+        self._grid_file = grid_file
+        self.grid = EclConfig._load_grid(grid_file)
+
+        self._refcase_file = refcase_file
+        self.refcase = EclConfig._load_refcase(refcase_file)
```

Comment:
Argument 1 to "_load_refcase" of "EclConfig" has incompatible type "Optional[str]"; expected "str"

Reply:
also fixed!